src/internal_telemetry/allocations/allocator/tracing_allocator.rs

```diff
@@ -59,10 +57,6 @@ unsafe impl<A: GlobalAlloc, T: Tracer> GlobalAlloc for GroupedTraceableAllocator
         try_with_suspended_allocation_group(
             #[inline(always)]
             |group_id| {
-                // We only set the group ID in the wrapper header if we're tracing an allocation, because when it
-                // comes back to us during deallocation, we want to skip doing any checks at all if it's already
-                // zero.
-                //
                 // If we never trace the allocation, tracing the deallocation will only produce incorrect numbers,
                 // and that includes even if we just used the rule of "always attribute allocations to the root
                 // allocation group by default".
```

Comment:
Can you also remove this too? It doesn't make a lot of sense without the previous paragraph that came before it.